backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -32,12 +69,26 @@
    * @return A list of member's engagements
    */
   public List<EngagementDto> getStaffEngagementsBetween(int staffId, Instant start, Instant end) {
-    Instant startTime = start == null ? Instant.MIN : start;
-    Instant endTime = end == null ? Instant.MAX : end;
     return engagementRepository.findByStaffId(staffId).stream()
```

Comment:
How do we think performance will be affected when someone has worked somewhere for a few years? Could be lots of engagements, will this cause issues? Just curious.

Reply:
Possibly, depends on company's retention policy for engagements.
Unfortunately, I can't think of another way of writing this at the moment.

Reply:
Yeah neither can I.... I don't know enough about databases to know the performance hit.

The only think I can think is that the generated SQL for this will end up always getting all engagement records from the database and _then_ performing a filter. Is there a way to construct an SQL statement that would avoid this? It's probably not part of MVP to get this bit right though.

Reply:
It should be possible to construct an sql statement which returns engagements between the dates.
I'll look into this, not sure if I'll make it part of this ticket though.